### PR TITLE
Test Quinn API

### PR DESCRIPTION
**Pull request by djc**

### CODE REVIEW

File: quinn/src/lib.rs [hunk 1]
```diff
@@ -1599,3 +1599,137 @@ impl<T: Read> Future for ReadToEnd<T> {
...
+                        let conn = conn.connection;
+                        let stream = conn.open_bi();
+                        stream
+                            .map_err(|_| ())
+                            .and_then(move |stream| {
+                                tokio::io::write_all(stream, b"foo".to_vec()).map_err(|_| ())
+                            })
+                            .and_then(|(stream, _)| tokio::io::shutdown(stream).map_err(|_| ()))
+                            .and_then(move |stream| {
+                                tokio::io::read_to_end(stream, Vec::new()).map_err(|_| ())
```
Comment (Ralith) [inline quinn/src/lib.rs]: Would it be appropriate to use the `read_to_end` defined in this module here?

File: quinn/src/lib.rs [hunk 2]
```diff
@@ -1599,3 +1599,137 @@ impl<T: Read> Future for ReadToEnd<T> {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
```
Comment (Ralith) [inline quinn/src/lib.rs]: It'd be nice for this to live in a separate file, I think.

File: quinn/src/lib.rs (hunk 1 above)
Comment (djc) [reply quinn/src/lib.rs]: Yup!

File: quinn/src/lib.rs (hunk 2 above)
Comment (djc) [reply quinn/src/lib.rs]: Agreed.

**Comment by djc** [comment]
Two reasons to do it now: nice coverage bump, and some automated testing in place before I can dive into multi-threaded endpoints.

**Pull request merged**